Meta/container.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+# Sets up a Docker container from which you can both build and run Ladybird
+
+set -e
+
+ARG0=$0
+print_help() {
+    NAME=$(basename "$ARG0")
+    cat <<EOF
+Usage: $NAME [COMMAND]
+       Sets up ephemeral Docker containers for Ladybird dev
+  Examples:
+    $NAME
+        Runs "./Meta/ladybird.sh run ladybird" inside the container
+    $NAME bash
+        Opens a bash shell inside the container
+    $NAME <cmd>
+        Runs arbitrary commands inside the container (from the Ladybird source tree)
+    $NAME docker
+        Sets up the Docker container/updates it if already installed
+    $NAME help
+        Shows this help screen
+EOF
+}
+
+usage() {
+    >&2 print_help
+    exit 1
+}
+
+docker_setup() {
+    local should_upgrade="$1"
+    
+    OLDID=$(docker images ladybird-dev -q)
+    if [ -z "${OLDID}" ]; then
+        echo "Building image from Dockerfile..."
+        docker build -f ./Meta/Dockerfile . -t ladybird-dev
+    elif [ "$should_upgrade" = "true" ]; then
+        echo "Removing old image..."
+        docker image rm ladybird-dev:latest
+        echo "Building new image from Dockerfile..."
+        docker build -f ./Meta/Dockerfile . -t ladybird-dev
+    fi
+}
+
+CMD=$1
+if [ "$CMD" = "help" ]; then
+    print_help
+    exit 0
+elif [ "$CMD" = "docker" ]; then
+    docker_setup "true"
+    exit 0
+fi
+
+ALL_CMDS=$*
+if [ -z "${ALL_CMDS}" ]; then
+    ALL_CMDS="./Meta/ladybird.sh run ladybird"
+fi
+
+# Build docker image if not installed already
+docker_setup "false"
+
+# Pass in Wayland socket from host to container, so you can run the browser
+# from the container itself
+
+TMP_DIR=$(mktemp --directory)
+
+docker run --rm -it \
```

Comment:
I saw there was some discussion on discord about this, but it seems I was asleep at the time :)

So, I'm not sure what this gives us over the devcontainer. The devcontainer is something that can be built and run through `VsCode`, or other IDEs that support the protocol. https://containers.dev/supporting

At first glance, it looks like it also forwards a wayland socket? 

I think that any solution involving making a docker-based workflow work better should be trying to knock off some tasks from https://github.com/LadybirdBrowser/ladybird/issues/318 rather than introducing a new workflow.

Reply:
Ah alright. I don't use an IDE, so this seemed a bit less interesting, and I thought it'd be nice to have a vanilla Docker image with the stuff baked in. However, since I already went to the trouble of getting all the dependencies in one place, it seems I can just update the dev container with my Dockerfile. 

The Wayland socket forwarding is the only new thing this gives us over the devcontainer I guess, but maybe that can also be added in? I dunno, someone who uses devcontainers would likely know more.